File: app/schemas/content.py
"""Pydantic schemas for content."""
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List

class ContentPublishStatusResponse(BaseModel):
    """Schema for individual target publish status (Meta, LinkedIn, etc.)."""
    meta_page_id: Optional[int] = None
    linkedin_account_id: Optional[int] = None
    status: str
    platform_post_id: Optional[str] = None
    error_message: Optional[str] = None

    class Config:
        from_attributes = True

class ContentBase(BaseModel):
    """Base content schema."""
    title: str = Field(..., min_length=1, max_length=200)
    body: str = Field(..., min_length=1)

class ContentCreate(ContentBase):
    """Schema for creating content."""
    organization_id: Optional[int] = Field(None, description="Optional: organization ID for team collaboration")
    schedule_at: Optional[datetime] = Field(None, description="Optional: when to publish (UTC); used after approval")
    schedule_meta_page_id: Optional[int] = Field(None, description="Optional: page to publish to when schedule_at is set")
    media_id: Optional[int] = Field(None, description="Optional: associated media file ID")

class ContentUpdate(BaseModel):
    """Schema for updating content."""
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    body: Optional[str] = Field(None, min_length=1)
    media_id: Optional[int] = Field(None, description="Optional: new media ID")
    organization_id: Optional[int] = Field(None, description="Optional: move content to another organization")

class ContentResponse(ContentBase):
    """Schema for content response."""
    id: int
    status: str
    organization_id: Optional[int]
    created_by_id: int
    approved_by_id: Optional[int]
    created_at: datetime
    updated_at: Optional[datetime]
    approved_at: Optional[datetime]
    schedule_at: Optional[datetime] = None
    schedule_meta_page_id: Optional[int] = None
    media_id: Optional[int] = None
    publish_statuses: List[ContentPublishStatusResponse] = []

    class Config:
        from_attributes = True

class ContentApprovalRequest(BaseModel):
    """Schema for approval/rejection request."""
    approved: bool
    comment: Optional[str] = None

class PublishToFacebookRequest(BaseModel):
    """Schema for publishing content to multiple Facebook Pages."""
    meta_page_ids: List[int]

class PublishToLinkedInRequest(BaseModel):
    """Schema for publishing content to multiple LinkedIn Accounts."""
    linkedin_account_ids: List[int]

class InsightsResponse(BaseModel):
    """Schema for post insights response."""
    reach: int
    engagement: int

File: frontend/src/api/content.ts
import { apiGet, apiPost, apiPatch, apiDelete } from './client';

export interface ContentPublishStatus {
  meta_page_id?: number | null;
  linkedin_account_id?: number | null;
  status: string; // 'pending' | 'processing' | 'posted' | 'failed'
  platform_post_id?: string | null;
  error_message?: string | null;
}

export interface Content {
  id: number;
  title: string;
  body: string;
  status: string;
  created_by_id: number;
  approved_by_id: number | null;
  created_at: string;
  updated_at: string | null;
  approved_at: string | null;
  schedule_at?: string | null;
  schedule_meta_page_id?: number | null;
  media_id?: number | null;
  media?: {
    url: string;
    mime_type: string;
  } | null;
  publish_statuses: ContentPublishStatus[];
}

export interface ContentCreate {
  title: string;
  body: string;
  schedule_at?: string | null;
  schedule_meta_page_id?: number | null;
  media_id?: number | null;
}

export interface ContentUpdate {
  title?: string;
  body?: string;
  media_id?: number | null;
}

export interface ApprovalRequest {
  approved: boolean;
  comment?: string;
}

export function listContent(params?: { skip?: number; limit?: number; status?: string; organization_id?: number }): Promise<Content[]> {
  return apiGet<Content[]>('content/', params as Record<string, string | number | undefined>);
}

export function getContent(id: number): Promise<Content> {
  return apiGet<Content>(`content/${id}`);
}

export function createContent(data: ContentCreate): Promise<Content> {
  return apiPost<Content>('content/', data);
}

export function updateContent(id: number, data: ContentUpdate): Promise<Content> {
  return apiPatch<Content>(`content/${id}`, data);
}

export function submitForApproval(id: number): Promise<Content> {
  return apiPost<Content>(`content/${id}/submit`);
}

export function approveContent(id: number, data: ApprovalRequest): Promise<Content> {
  return apiPost<Content>(`content/${id}/approve`, data);
}

export function deleteContent(id: number): Promise<void> {
  return apiDelete(`content/${id}`);
}

export function publishToFacebook(contentId: number, metaPageIds: number[]): Promise<Content> {
  return apiPost<Content>(`content/${contentId}/publish-to-facebook`, { meta_page_ids: metaPageIds });
}

export function publishToLinkedIn(contentId: number, linkedinAccountIds: number[]): Promise<Content> {
  return apiPost<Content>(`content/${contentId}/publish-to-linkedin`, { linkedin_account_ids: linkedinAccountIds });
}

export interface Insights {
  reach: number;
  engagement: number;
}

export function getInsights(contentId: number, metaPageId: number): Promise<Insights> {
  return apiGet<Insights>(`content/${contentId}/insights`, { meta_page_id: metaPageId });
}
